src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -21,14 +23,23 @@ import scala.util.{Try, Failure, Success}
 object ValidateActor {
   private val tag = "ValidateActor"
 
-  sealed trait ValidateActorMessage
-  case object ValidateWorkflow extends ValidateActorMessage
-
   def props(wdlSource: WdlSource, wdlJson: Option[WdlJson], workflowOptions: Option[WdlJson]): Props = {
     Props(new ValidateActor(wdlSource, wdlJson, workflowOptions))
   }
+
+  sealed trait ValidateActorMessage
+  sealed trait ValidationResult extends ValidateActorMessage
+  //This is the reply that this actor sends in case of a succsessful validation. It contains all the resolved / validated components of
+  //the workflow (minus Declarations). The callee should reuse these materilized information.
+  case class ValidationSuccess(namespaceWithWorkflow: NamespaceWithWorkflow,
+                               coercedInputs: Option[WorkflowCoercedInputs],
+                               workflowOptions: Option[WorkflowOptions],
+                               runtimeAttributes: Seq[Set[String]]) extends ValidationResult
+  case class ValidationFailure(reason: Throwable) extends ValidationResult
```

Comment:
(genuine question :-) - If this is an async message, how does the receiver know which validation has failed?